python-avd/pyavd/_cv/client/__init__.py

```diff
@@ -72,10 +103,56 @@ def _connect(self) -> None:
         self._set_version()
 
         if self._channel is None:
-            self._channel = Channel(host=self._servers[0], port=self._port, ssl=ssl_context)
+            if self._proxy_manager is not None:
+                self._channel = await self._create_proxy_channel(ssl_context)
+            else:
+                self._channel = Channel(host=self._servers[0], port=self._port, ssl=ssl_context)
 
         self._metadata = {"authorization": "Bearer " + self._token}
 
+    async def _create_proxy_channel(self, ssl_context: ssl.SSLContext | bool) -> Channel:
+        """
+        Create a gRPC channel using the proxy manager.
+
+        Args:
+            ssl_context: SSL context for destination server connection.
+
+        Returns:
+            Configured gRPC Channel instance.
+        """
+        if self._proxy_manager is None:
+            msg = "Proxy manager not initialized"
+            raise CVClientException(msg)
+
+        # Create the channel first
+        channel = Channel(host=self._servers[0], port=self._port, ssl=ssl_context)
+
+        # Create custom connector that uses proxy
+        async def proxy_connection() -> asyncio.Protocol:
+            loop = asyncio.get_running_loop()
+
+            try:
+                # Create socket through proxy using python-socks
+                proxy_sock = await self._proxy_manager.create_socket_for_grpc()
+
+                # Create the gRPC protocol using the proxy socket
+                transport, protocol = await loop.create_connection(
+                    lambda: channel._protocol_factory(),
+                    sock=proxy_sock,
+                    ssl=ssl_context,
+                    server_hostname=self._servers[0] if ssl_context else None,
+                )
+
+            except Exception as e:
+                msg = f"Failed to create proxy connection: {type(e).__name__}: {e}"
+                raise CVClientException(msg) from e
+
+            else:
+                return protocol
+
+        channel._create_connection = proxy_connection
```

Comment:
We should probably hardcode our requirement for grpclib so any changes there won't break this.

```suggestion
        # Override the standard method from grpclib with our proxy variant.
        channel._create_connection = proxy_connection
```

Reply:
* added comment
* hardcoded "grpclib==0.4.8" as pyavd direct dep